### PR TITLE
[1/3] Implement basic ride model 

### PR DESCRIPTION
**Pull request by katesliang**
## Overview

Implement basic ride model

## Changes Made

- Created ride model (ride details)
- Created path model (start and end location names and id)

## Test Coverage

N/A

## Next Steps

Implementing ride serializer in next PR

### CODE REVIEW

File: src/path/models.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from django.db import models
+
+
+class Path(models.Model):
+    start_location_id = models.TextField()
```
Comment (noah-solomon) [inline src/path/models.py]: wouldn't this make sense as an integer? also what does the id refer to?

File: src/path/models.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    start_location_name = models.TextField()
+    end_location = models.TextField()
```
Comment (noah-solomon) [inline src/path/models.py]: `end_location_id` for consistency, also same question about being an integer

**Review by noah-solomon — APPROVED** [review]
looks valid to me, just a few questions

File: src/path/models.py (hunk 1 above)
Comment (katesliang) [reply src/path/models.py]: frontend is using google places api and id is a string: https://developers.google.com/maps/documentation/places/web-service/place-id 

will add comments to make this clearer :) tyyy

File: src/path/models.py (hunk 2 above)
Comment (katesliang) [reply src/path/models.py]: omg nice catch

File: src/path/models.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from django.db import models
+
+
+class Path(models.Model):
+    start_location_place_id = models.TextField()  # start location ID from Google Places
```
Comment (alanna-zhou) [inline src/path/models.py:5]: can you not do a char field? you're using up a lot of space for a textfield

File: src/ride/models.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+from person.models import Person
+
+
+class Ride(models.Model):
+    creator = models.ForeignKey(
+        Person, on_delete=models.SET_NULL, null=True, related_name="creator"
+    )
+    max_travelers = models.IntegerField(default=1)  # max number of travelers
+    min_travelers = models.IntegerField(default=1)  # min number of travelers
+    datetime = models.DateTimeField()  # date and time of travel
```
Comment (alanna-zhou) [inline src/ride/models.py]: start_datetime?

File: src/ride/models.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    description = models.TextField(default=None, null=True)
+    driver = models.ForeignKey(
+        Person, on_delete=models.SET_NULL, null=True, related_name="driver"
+    )
+    is_flexible = models.BooleanField(default=False)
+    riders = models.ManyToManyField(Person)
+    estimated_cost = models.FloatField(default=None, null=True)
+    path = models.OneToOneField(Path, on_delete=models.CASCADE, primary_key=True)
+    RIDE_TYPES = [("rideshare", "Rideshare"), ("studentdriver", "Student Driver")]
```
Comment (alanna-zhou) [inline src/ride/models.py]: put this right under `class Ride...` cuz it's an enum type

File: src/ride/models.py (hunk 4 above)
Comment (alanna-zhou) [reply src/ride/models.py]: i feel like `departure_datetime` is good

**Review by alanna-zhou — APPROVED** [review]
if you address you can merge

File: src/path/models.py (hunk 3 above)
Comment (katesliang) [reply src/path/models.py:5]: i chose textfield bc they said "there is no maximum length for Place IDs" in api doc :(

Comment (katesliang) [reply src/path/models.py:5]: ill probs change place name to max char length of 100?

Comment (alanna-zhou) [reply src/path/models.py:5]: i see.. ahh i guess textfield is fine then for both

Comment (katesliang) [reply src/path/models.py:5]: okie tyyy

**Pull request merged**